metricbeat/module/system/filesystem/helper_test.go

```diff
@@ -47,6 +47,103 @@ func TestFileSystemList(t *testing.T) {
 	}
 }
 
+func TestFileSystemListFiltering(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("These cases don't need to work on Windows")
+	}
+
+	cases := []struct {
+		description   string
+		fss, expected []sigar.FileSystem
+	}{
+		{
+			fss: []sigar.FileSystem{
+				{DirName: "/", DevName: "/dev/sda1"},
+				{DirName: "/", DevName: "/dev/sda1"},
+			},
+			expected: []sigar.FileSystem{
+				{DirName: "/", DevName: "/dev/sda1"},
+			},
+		},
+		{
+			description: "Don't repeat devices, sortest of dir names should be used",
```

Comment:
nit: shortest, also in all the ones below

Reply:
:D